### PR TITLE
Resolving nested obligations

### PR DESCRIPTION
**Pull request by franzliedke**
Wrote a failed test. Will work on the implementation.

### CODE REVIEW

**Comment by jgraichen** [comment]
:+1:

**Comment by franzliedke** [comment]
Now we have a working implementation. Not sure whether this is the safest.

I also suspect it will fail when the block passed to `Dependent` expects multiple arguments.

File: lib/obligation/impl.rb [hunk 1]
```diff
@@ -149,6 +145,18 @@ def _value(timeout)
...
     end
+
+    def _resolved_dependencies
+      return @dependencies.map(&:value) if @dependencies.is_a?(Array)
+
+      @dependencies.value
+    end
+
+    def _nested(result)
+      return result.value if result.respond_to? :value
```
Comment (franzliedke) [inline lib/obligation/impl.rb:156]: Using `result.respond_to? :then` may be a better check here.

**Comment by franzliedke** [comment]
Alternative implementation: Replacing `Obligation::Base#then` with the following:

``` ruby
def then
  return self unless block_given?

  handler = Proc.new
  Dependent.new(self, &lambda { |val|
    res = handler.call(val)
    res.respond_to?(:then) ? res.value : res
  })
end
```

This is closer to how jQuery does it, for example, but shouldn't matter much in the end. Your choice. :)

**Pull request merged**